### ISSUE TITLE
Hotload Dim spam since update

### ISSUE THREAD

**Issue by Sir-Will** [DarkholmeTenk/TardisModIssues#436]
hey,

after updating from tardismod-1.7.10-0.99-207 to tardismod-1.7.10-0.994-124-DS it started to spam with this error:

```
 Cannot Hotload Dim: Could not get provider type for dimension 959, does not exist
```

Downgrading the mod stops the error spam.

**Comment by FoxPotato** [comment]
Could you maybe provide us with your mod list?

**Comment by Sir-Will** [comment]
https://github.com/MyM-ModpackTeam/GalacticScience/tree/master/src/mods

**Comment by DarkholmeTenk** [comment]
Try build 91 of darkcore, also is anything happening when this happens? any players online? anything like that?

**Comment by Sir-Will** [comment]
I will test build 91 later.
It requires at least one player join after an restart that it starts.

**Comment by DarkholmeTenk** [comment]
Is that person in a tardis? Could you upload the output of "/treg list" too pls and a full console excerpt from around the time it happens rather than just that line pls.

**Comment by Sir-Will** [comment]
Same issue with build 91.
No, not in tarids.

```
Dimension mapping:
biteboy5->772
Kazuto_Senpai->773
YiayiasBoy->1029
MidnightNinjaBoi->774
JakeHillKing->775
GiaToai->903
ChetwyndKid->523
Omega_Inferno->780
VoidPanCake->526
hoothootmen->782
Amped_Gamin->1038
MCSGB->1040
shafnowab->786
xd_antonio_xd2->787
Bandit226->1043
darkmonger12->1044
fedprod->789
Butterfilms101->1173
Jacuza->662
SpeederSam->1432
BioChip->2076
caithleen->159
CatheteriZedEYE->927
stoxy999->928
_ZerefDragneel_->1058
DavetStone92->803
SirWill->164
korncover->1317
harryboase->938
A_Super_Creeper->1066
maeamian->567
Legallity->1463
Barefootfox1663->824
Umaroth17->1208
Emberlore->830
Xicer1->1091
SenNox21->1987
malachix7->1092
simfrek->718
shinks45->1102
theshadowpanther->719
logjohn9->975
SpaceEll->1103
Vladimir_Put_In->976
Vampiredawn->721
strayturtles->1107
Sikeslayer->724
Log_->997
dungeonax->1003
RockinVSGaming->1261
_Ianite_->1517
Zarrexnet->754
Jarl_Giga->627
ProBattles->755
6ustrik->2036
thomkileur->2038
appollon41->2039
theMineMaster2->888
LoffeLoek->890
Dragon59MC->1019
Dylan5124->1020
AzuleAdos->1404
```

**Comment by DarkholmeTenk** [comment]
Have you perchance used "/treg remove total" at all?

**Comment by Sir-Will** [comment]
No, what does that command?

**Comment by DarkholmeTenk** [comment]
Try latest darkcore and tardismod from the jenkins
Try "/treg listdims" too if it doesn't stop :P

**Comment by Sir-Will** [comment]
Still shows it with the latest version.

```
Dims: 772,773,1029,774,775,903,523,780,526,782,1038,1040,786,787,1043,1044,789,1173,662,1432,2076,159,927,928,1058,803,164,1317,938,1066,567,1463,824,1208,830,1091,19
```

**Comment by DarkholmeTenk** [comment]
I have no idea, unless there's more info in the console, it'll be very hard to determine, since it just looks like you don't have a dimension registered to dim 959, so I don't know why anything is referring to it :P

**Comment by Sir-Will** [comment]
Do you save the registered world's somewhere? 
I wonder how it gets the dim id if there is no folder for it.

**Comment by FoxPotato** [comment]
In the save folder, there's another folder called tardis. Take a look in there :P

**Comment by Sir-Will** [comment]
@FoxPotato I already checked that, there is no DIM959 folder. If the mod doesn't save anything else somewhere else it is really weird.
I have to downgrade the mod and stay on the version then :/

**Comment by DarkholmeTenk** [comment]
Some data will be in the world's data folder, but if there's no dim959 folder, it suggests it's not a tardis dimension, so I have absolutely no idea

**Comment by Sir-Will** [comment]
So, after some very long debugging/testing I found out that deleting the tardis 938 stopped the spam I mentioned above but this showed another issue.
It keeps loading/unloading dimensions and this does only happen with tardismod-1.7.10-0.994-124 and above, no issue with tardismod-1.7.10-0.99-207.

```
[14:32:17] [Server thread/INFO] [FML/]: Unloading dimension 28
[14:32:17] [Server thread/INFO] [FML/]: Unloading dimension 7
[14:32:17] [Server thread/INFO] [FML/]: Loading dimension 28 (world) (net.minecraft.server.dedicated.DedicatedServer@4fb93043)
[14:32:17] [Server thread/INFO] [FML/]: Loading dimension 7 (world) (net.minecraft.server.dedicated.DedicatedServer@4fb93043)
```

The messages only show up if chunks get loaded. (Flying/walking around)

Any idea what change causes this in your mod?

**Comment by FoxPotato** [comment]
Are there perhaps TARDISes in those dimensions? Not that that's a reason for them to load/unload, but it would atleast make it easier to find where in the code the problem is :P

**Comment by Sir-Will** [comment]
28 and 7 is a space station from galacticraft.

**Comment by FoxPotato** [comment]
And you said it's just with a specific version of tardis mod?
Are you sure it's not galacticraft?

**Comment by Sir-Will** [comment]
Why would it be galacticraft if I have no issues after downgrading tardis mod?

**Comment by FoxPotato** [comment]
Could be a combination of the two. Are there TARDISes present in those dimensions?
Perhaps tardismod thinks those dimensions are TARDIS dimensions and tries to keep them loaded. I don't actually know what dark changed between the 2 versions you're using so i can't really fix this :/

**Comment by DarkholmeTenk** [comment]
Fair point. Do "/dcd cl list" and check if any of the chunk loaders are supposedly there

**Comment by Sir-Will** [comment]
I somehow can no longer reproduce the dimension load/unload issue.
I'm not sure why it does no longer happen.

**Comment by FoxPotato** [comment]
Well, maybe it was indeed tardismod thinking those dimensions were console rooms. But if it's fixed would you could close this issue.
And if you were to come across the same bug again, open it up.

**Comment by Sir-Will** [comment]
Issue just started again with a different world :(
`Cannot Hotload Dim: Could not get provider type for dimension 2311, does not exist`

**Comment by DarkholmeTenk** [comment]
Use the latest darkcore/tardismod. Turn on debug text in darkcore and show us the stack trace.

**Comment by Sir-Will** [comment]
Just updated to the latest versions and I no longer can reproduce the issue. Looks like you already fixed it with your latest changes.

**Comment by Sir-Will** [comment]
Still an Issue in tardismod-1.7.10-0.995-350

**Comment by DarkholmeTenk** [comment]
In that case, my last comment stands.

**Comment by Sir-Will** [comment]
Debug doesn't show anything while it spams the "Cannot Hotload Dim" message and it only spams it if I'm in the overworld.

**Comment by DarkholmeTenk** [comment]
If nothing is mentioned with Darkcore's debug mode on, then none of my code is trying to load a dimension so I don't believe this is my mod's fault.

**Comment by einsteinK** [comment]
I got this error too after using /tremove (or /trepair? I forgot) (might be with parameter "total" eh).
It seems like the dimension isn't completely deleted, but there's no WorldProvider for it.
That only shows that explicitly deleting a TARDIS leads to a broken dimension, not by updating.
(No idea if this information helps, but maybe it does, so eh)

**Comment by Sir-Will** [comment]
@DarkholmeTenk can you explain me then why the issue is gone if I remove the tardis mod?

**Comment by DarkholmeTenk** [comment]
Does it stay if you keep darkcore but not TM?

**Comment by Sir-Will** [comment]
I only removed tardis mod and the issue was gone, darkcore was still enabled.

**Comment by DarkholmeTenk** [comment]
What steps reproduce the problem? Deleting folders or anything?

Is there any sort of stack trace, with darkcore's debug mode on?
Is this on a dedicated server, client side or single player?

Was dimension 2311 ever a TARDIS dimension?
Was dimension 2311 from another mod which has been removed?
